### PR TITLE
[Service] Fix Support `text/csv` content type in request bodies in ballerina service generation

### PR DESCRIPTION
**Pull request by lnash94**
- Fix Support `text/csv` content type in request bodies and response in ballerina service generation
- Fix NPE issue when response content is empty
- Fix NPE issue when yaml content include http status codes which are not supported in ballerina http (ex: 429) https://github.com/ballerina-platform/openapi-tools/issues/819

## Automation tests
 - Unit tests  - add
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ServiceDiagnosticMessages.java [hunk 1]
```diff
@@ -45,7 +45,9 @@
             DiagnosticSeverity.ERROR),
     OAS_SERVICE_106("OAS_SERVICE_106",
             "Header '%s' with no header type can not be mapped to the Ballerina headers.",
-            DiagnosticSeverity.ERROR);
+            DiagnosticSeverity.ERROR),
+    OAS_SERVICE_107("OAS_SERVICE_107", "Http status code '%s' won't support in Ballerina.",
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ServiceDiagnosticMessages.java:49]:
```suggestion
    OAS_SERVICE_107("OAS_SERVICE_107", "HTTP status code '%s' is not supported in Ballerina.",
```

File: openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ReturnTypeGenerator.java [hunk 2]
```diff
@@ -219,7 +220,12 @@ private UnionTypeDescriptorNode getUnionNode(Iterator<Map.Entry<String, ApiRespo
         while (responseIter.hasNext()) {
             Map.Entry<String, ApiResponse> response = responseIter.next();
             String code = GeneratorConstants.HTTP_CODES_DES.get(response.getKey().trim());
-            if (response.getValue().getContent() == null && response.getValue().get$ref() == null) {
+            if (code == null && !response.getKey().trim().equals("default")) {
+                throw new BallerinaOpenApiException(String.format(OAS_SERVICE_107.getDescription(),
+                        response.getKey().trim()));
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ReturnTypeGenerator.java:225]: Lets convert `response.getKey().trim()` into a local variable and reuse.

File: openapi-cli/src/test/java/io/ballerina/openapi/generators/service/RealAPITests.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * All the tests related to the {@code io.ballerina.openapi.generators.service.ReturnTypeGenerator} util.
+ */
+public class RealAPITests {
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/java/io/ballerina/openapi/generators/service/RealAPITests.java:37]: WDYM by `RealAPITests`?

File: openapi-cli/src/test/java/io/ballerina/openapi/generators/service/RealAPITests.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private static final Path RES_DIR = Paths.get("src/test/resources/generators/service").toAbsolutePath();
+    List<String> list1 = new ArrayList<>();
+    List<String> list2 = new ArrayList<>();
+    Filter filter = new Filter(list1, list2);
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/java/io/ballerina/openapi/generators/service/RealAPITests.java:41]:
```suggestion
    Filter filter = new Filter(new ArrayList<>(), new ArrayList<>());
```

**Review by NipunaRanasinghe — CHANGES_REQUESTED** [review]

File: openapi-cli/src/test/java/io/ballerina/openapi/generators/service/RealAPITests.java (hunk 3 above)
Comment (lnash94) [reply openapi-cli/src/test/java/io/ballerina/openapi/generators/service/RealAPITests.java:37]: I created this test class to test the real-world API scenarios

Comment (NipunaRanasinghe) [reply openapi-cli/src/test/java/io/ballerina/openapi/generators/service/RealAPITests.java]: @lnash94 So all the other tests are not real world scenarios? ;)

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**